esp-hal/src/uart.rs

```diff
@@ -594,9 +625,15 @@ where
         self.regs().status().read().txfifo_cnt().bits().into()
     }
 
-    /// Flush the transmit buffer of the UART
+    /// Flush the transmit buffer.
+    ///
+    /// This function blocks until all data in the TX FIFO has been
+    /// transmitted.
     #[instability::unstable]
     pub fn flush(&mut self) -> Result<(), TxError> {
+        while self.tx_fifo_count() > 0 {}
+        #[cfg(esp32c2)]
+        crate::rom::ets_delay_us(10);
```

Comment:
I'm not sure why this isn't an issue on other chips, but we can't tell whether the FSM is idle _before_ sending out the last byte, or _after_ sending out the last byte. 10us seems to be enough with the slowest baud rates, although it's a tad bit wasteful @ 5Mbaud/s

Reply:
> ~~I'm not sure why this isn't an issue on other chips~~, (it is an issue on other chips)

If it is an issue on other chips, why are we only doing this for the c2?

FWIW, I think the delay is fine, most use cases will just keep pushing data into `write` and then do a final flush at the end - hopefully not a big deal.